### PR TITLE
New system sounds

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose

Implement system sounds.

JIRA issue: [CORE-13951](https://jira.reactos.org/browse/CORE-13951)

## Proposed changes

- Replacement for PR [#4093](https://github.com/reactos/reactos/pull/4093) and [#4094](https://github.com/reactos/reactos/pull/4094)
- Add converted sound files

### Working sounds
- MessageBox and MessageBeep sounds
- Empty recycle bin sound (Improved)
- Logoff sound

## TODO

Nothing to do, built and tested.

### CODE REVIEW

**Comment by TAN-Gaming** [comment]
https://user-images.githubusercontent.com/86486978/147526983-6244e074-1cd2-4993-bf2a-a7a73b1558e9.mp4

**Comment by binarymaster** [comment]
You probably want to close referenced PRs in description, as this one supersedes them.

**Comment by HBelusca** [comment]
Logon sound is a bit too long... if I were you I would shorten (I mean, make it faster) the fading-away at the end.

**Comment by oleg-dubinskiy** [comment]
> Logon sound is a bit too long...

Indeed, and it doesn't fit to other sounds.
Also please note that there are also other sound schemes made by other contributors:
https://reactos.org/wiki/User:Petr-akhlamov/ReactOS_sound_schemes.
So we may want to make a voting which one from them is better (and hence, which will be used).
Or, as an option, just keep the current logon sound for now, because: 1) it fits to other sounds theme, 2) it was chosen by community via the voting.

**Comment by TAN-Gaming** [comment]
> Logon sound is a bit too long... if I were you I would shorten (I mean, make it faster) the fading-away at the end.

I used it for testing only, not included in this PR.

File: boot/bootdata/hivedef.inf [hunk 1]
```diff
@@ -2046,14 +2046,14 @@ HKCU,"AppEvents\Schemes\Apps\.Default\SystemAsterisk",,0x00000012
 HKCU,"AppEvents\Schemes\Apps\.Default\SystemAsterisk\.Current","",0x00020000,""
 HKCU,"AppEvents\Schemes\Apps\.Default\SystemAsterisk\.Default","",0x00020000,""
 HKCU,"AppEvents\Schemes\Apps\.Default\SystemExclamation",,0x00000012
-HKCU,"AppEvents\Schemes\Apps\.Default\SystemExclamation\.Current","",0x00020000,""
-HKCU,"AppEvents\Schemes\Apps\.Default\SystemExclamation\.Default","",0x00020000,""
+HKCU,"AppEvents\Schemes\Apps\.Default\SystemExclamation\.Current","",0x00020000,"%SystemRoot%\media\ReactOS_Notice.wav"
```
Comment (oleg-dubinskiy) [inline boot/bootdata/hivedef.inf:2049]: Please revert this and other registry entries. We don't always need to set them. They are automatically set at 2nd setup stage for Workstation configuration if the appropriate sound file exists. But for Server, we keep the default behaviour, without sounds enabled, excepting logon sound. Same as in Windows Server 2003 SP2. See the commit: cbf8b367f5873f8ed9bc558326e8270717d58f45.

File: media/CMakeLists.txt [hunk 2]
```diff
@@ -1,18 +1,12 @@
 
+add_subdirectory(drivers)
```
Comment (oleg-dubinskiy) [inline media/CMakeLists.txt:2]: This change is unrelated to the purpose of this PR and does not make much sense. The configuration files in drivers/etc are not related to sound events at all. Please revert it (and remove the corresponding CMake file you've added).

**Review by oleg-dubinskiy — COMMENTED** [review]
Some nit picks. In other it's fine. :slightly_smiling_face:

File: base/system/winlogon/sas.c [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * FILE:            base/system/winlogon/sas.c
  * PURPOSE:         Secure Attention Sequence
  * PROGRAMMERS:     Thomas Weidenmueller (w3seek@users.sourceforge.net)
- *                  Herv� Poussineau (hpoussin@reactos.org)
+ *                  Hervé Poussineau (hpoussin@reactos.org)
```
Comment (learn-more) [inline base/system/winlogon/sas.c:7]: Encoding change, please fix.

File: base/system/winlogon/sas.c [hunk 4]
```diff
@@ -1296,7 +1104,7 @@ DispatchSAS(
                 if (bSecure)
                 {
                     wlxAction = WLX_SAS_ACTION_LOCK_WKSTA;
-//                    DoGenericAction(Session, WLX_SAS_ACTION_LOCK_WKSTA);
+                    // DoGenericAction(Session, WLX_SAS_ACTION_LOCK_WKSTA);
```
Comment (learn-more) [inline base/system/winlogon/sas.c:1107]: Unrelated change, please revert.

File: dll/win32/shell32/folders/CRecycleBin.cpp [hunk 5]
```diff
@@ -982,6 +982,119 @@ TRASH_TrashFile(LPCWSTR wszPath)
...
+    LPCWSTR lpRegSubKey = L"AppEvents\\Schemes\\Apps\\Explorer\\EmptyRecycleBin";
+    HKEY hRegKey = NULL;
+    HKEY hRegSnd = NULL;
+    LPWSTR lpRegVal = NULL;
+    LPWSTR lpSndPath = NULL;
+    DWORD dwRegValType, dwRegValSize, dwSndPathLen;
+    LONG lError;
+
+    /* Open registry key */
+    lError = RegOpenKeyExW(HKEY_CURRENT_USER, lpRegSubKey, 0, KEY_READ, &hRegKey);
```
Comment (learn-more) [inline dll/win32/shell32/folders/CRecycleBin.cpp:996]: This is a cpp file, no need for all this manual closing /error handling, `CRegKey` for the registry and `CHeapPtr<WCHAR>` for the buffers should cut about 70% off from this function.

File: base/system/winlogon/sas.c (hunk 3 above)
Comment (TAN-Gaming) [reply base/system/winlogon/sas.c:7]: What is the encoding I should use for this file?

Comment (learn-more) [reply base/system/winlogon/sas.c:7]: Whatever it was

**Review by HBelusca — CHANGES_REQUESTED** [review]
Please revert the last two commits regarding hivedef.inf, and add changes in it correctly.
And please stop using the GitHub interface as a file editor (the "Add files via upload" automatic GitHub commit message gives it away).

**Comment by TAN-Gaming** [comment]
> Please revert the last two commits regarding hivedef.inf, that break the file history, and add changes in it correctly instead. And please stop using the GitHub interface as a file editor (the "Add files via upload" automatic GitHub commit message gives it away). It's breaking the file encoding.

I can't fix it, so I closed it now.
